5.5/test/test_replication.sh

```diff
@@ -0,0 +1,145 @@
+#!/bin/bash
+
+set -eux
+
+TIME_SEC=1000
+TIME_MIN=$((60 * $TIME_SEC))
+HOST_DOCKER_IP=$(ip -4 addr show docker0 | grep -Po 'inet \K[\d.]+')
+OPENSHIFT_CONFIG_DIR="/tmp/openshift-config"
+OPENSHIFT_NODE_CONFIG="node-`hostname`/node-config.yaml"
+
+# time_now return the time since the epoch in millis
+function time_now()
+{
+  echo $(date +%s000)
+}
+
+# wait_for_url_timed attempts to access a url in order to
+# determine if it is available to service requests.
+#
+# $1 - The URL to check
+# $2 - Optional prefix to use when echoing a successful result
+# $3 - Optional maximum time to wait before giving up (Default: 10s)
+function wait_for_url_timed {
+  STARTTIME=$(date +%s)
+  url=$1
+  prefix=${2:-}
+  max_wait=${3:-10*TIME_SEC}
+  wait=0.2
+  expire=$(($(time_now) + $max_wait))
+  set +e
+  while [[ $(time_now) -lt $expire ]]; do
+    out=$(curl -k --max-time 2 -fs $url 2>/dev/null)
+    if [ $? -eq 0 ]; then
+      set -e
+      echo ${prefix}${out}
+      ENDTIME=$(date +%s)
+      echo "[INFO] Success accessing '$url' after $(($ENDTIME - $STARTTIME)) seconds"
+      return 0
+    fi
+    sleep $wait
+  done
+  echo "ERROR: gave up waiting for $url"
+  set -e
+  return 1
+}
+
+function setup_dns() {
+  # Make ourselves the default resolver. This is needes so FQDNs such as
+  # "pod-name.namespace.svc.cluster.local" can be resolved.
+  if ! grep -q $HOST_DOCKER_IP /etc/resolv.conf; then
+    sudo sed -i "1inameserver $HOST_DOCKER_IP" /etc/resolv.conf
+  fi
+
+  cat /etc/resolv.conf
+
+  sudo rm -rf $OPENSHIFT_CONFIG_DIR
+  # Generate openshift config file and edit the node config with DNS pointing to us.
+  docker run --rm -i --privileged --net=host \
+    -v $OPENSHIFT_CONFIG_DIR:/config \
+    openshift/origin start --write-config=/config
+  sudo sed -i "s/dnsIP: .*/dnsIP: $HOST_DOCKER_IP/" $OPENSHIFT_CONFIG_DIR/$OPENSHIFT_NODE_CONFIG
+}
+
+# Start openshift. We're using local generated config file that has DNS updated.
+function start_openshift() {
```

Comment:
ugh.  this is good work but also ugh (that we're adding all this logic to start/configure openshift in yet another location)